dashboard/lib/dynamic_config/dcdo.rb

```diff
@@ -0,0 +1,45 @@
+# A dynamic configuration module that allows us to update
+# config settings without pushing new code.
+
+require 'digest/sha1'
+require 'dynamic_config/datastore_cache'
+require 'dynamic_config/adapters/dynamodb_adapter'
+require 'dynamic_config/adapters/json_file_adapter'
+
+if CDO.use_dynamo_tables
```

Comment:
This code continues the "static" patterns used elsewhere in our code, using global configuration variables and module/class methods, which make it impossible to, say, plug in a new adapter (say an in-memory config adapter for tests) without modifying this file. 

Perhaps we could strike a compromise between using globals everywhere and requiring all callers to pass in dependencies?   For example, we could have a factor method that applies the default policies for passing in the dependencies and returns a default cached instance, but also an initializer that construct a new instance with arbitrary dependency implementations passed in.